scalafix-testkit/src/main/scala/scalafix/testkit/TestkitProperties.scala

```diff
@@ -21,17 +22,21 @@ import scala.meta.Classpath
 final class TestkitProperties(
     val inputClasspath: Classpath,
     val inputSourceDirectories: List[AbsolutePath],
-    val outputSourceDirectories: List[AbsolutePath]
+    val outputSourceDirectories: List[AbsolutePath],
+    val sourceRoot: AbsolutePath
 ) {
   def inputSourceDirectory: AbsolutePath =
     inputSourceDirectories.head
   def outputSourceDirectory: AbsolutePath =
     outputSourceDirectories.head
+  def inputOffset: RelativePath = inputSourceDirectory.toRelative(sourceRoot)
```

Comment:
Would it be possible for you to add the flag `-P:semanticdb:sourceroot:<path>` in pants to compile the input source directories? This is what the scalafix.g8 template generates https://github.com/scalacenter/scalafix.g8/blob/1bc1e8643666350d93b5096be4a7830bba36dc01/scalafix/build.sbt#L17-L20

Otherwise, it's actually a cleaner solution to use the default sourceroot and handle this offset in testkit instead. That would avoid the custom sbt config in every scalafix.g8 project 🤔

Reply:
Pants does pass -P:semanticdb:sourceroot correctly both when compiling and when calling scalafix-cli.

Reply:
To be more specific I believe the issue is that pants passes the sourceroot param, which causes semanticdb to store the semanticdb uri relative to a specific path, but that relative path offset is never passed to testkit.